### ISSUE TITLE
enhancement(content-schema): allow bidirectional transform cycles with inefficiency

### ISSUE THREAD

**Issue by hansjm10** [hansjm10/Idle-Game-Engine#610]
## Problem

The transform system has cycle detection that prevents bidirectional resource conversions - a common game mechanic:

```
Transform cycle detected: essence-to-gold → gold-to-essence → essence-to-gold
(involves resources: essence, gold).
```

While cycle detection is valuable for preventing infinite loops, it blocks legitimate game designs:
- Currency exchange (gold ↔ gems with fee)
- Material recycling (craft → scrap → partial materials)
- Energy conversion (mana ↔ stamina with loss)

## Current Workaround

- Remove one direction of conversion
- Use generators with production/consumption instead (less intuitive)
- Complex workarounds with intermediate resources

### Option 1: Allow with warning
```
Warning: Transform cycle detected. Ensure conversions are not net-positive.
```

### Option 2: Add opt-in flag
```typescript
{
  id: 'gold-to-essence',
  allowInCycle: true,  // Acknowledge the cycle is intentional
  inputs: [{ resourceId: 'gold', amount: 100 }],
  outputs: [{ resourceId: 'essence', amount: 80 }],  // 20% loss = not profitable
}
```

### Option 3: Detect profitable cycles only
Only error if the cycle is net-positive (would allow infinite resources).

## Recommendation

Option 3 is ideal but complex. Option 2 is a good middle ground - content authors explicitly acknowledge cycles are intentional.

## Acceptance Criteria

- [ ] Bidirectional conversions with net loss are allowed
- [ ] Profitable cycles (infinite resources) are still prevented
- [ ] Clear documentation of cycle rules
